### PR TITLE
fix: add missing untar flag to pull chart from oci registry

### PR DESCRIPTION
**Pull request by toVersus**
It seems that #290 introduced a bug and we no longer generate manifests from OCI based chart:

```console
=== RUN   TestHelmfileTemplateWithBuildCommand/oci_chart_pull
    snapshot_test.go:172: Output from [-f testdata/snapshot/oci_chart_pull/input.yaml template]: Pulling localhost:5000/myrepo/raw:0.1.0
        Pulled: localhost:5000/myrepo/raw:0.1.0
        Digest: sha256:4c321df86a2b822e62583a45f0e9507027e10b0aca232665b1cd6be6da0cdc19

        in testdata/snapshot/oci_chart_pull/input.yaml: [release "foo": command "/usr/local/bin/helm" exited with non-zero status:

        PATH:
          /usr/local/bin/helm

        ARGS:
          0: helm (4 bytes)
          1: pull (4 bytes)
          2: oci://localhost:5000/myrepo/raw (31 bytes)
          3: --version (9 bytes)
          4: 0.1.0 (5 bytes)
          5: --destination (13 bytes)
          6: /var/folders/j5/238lrgj54dz032np_n85c8nh0000gp/T/helmfile3497684751/foo/raw/0.1.0 (81 bytes)

        ERROR:
          exit status 1

        EXIT STATUS
          1

        STDERR:
          Error: open /var/folders/j5/238lrgj54dz032np_n85c8nh0000gp/T/helmfile3497684751/foo/raw/0.1.0/raw-0.1.0.tgz1281403981: no such file or directory

        COMBINED OUTPUT:
          Pulled: localhost:5000/myrepo/raw:0.1.0
          Digest: sha256:4c321df86a2b822e62583a45f0e9507027e10b0aca232665b1cd6be6da0cdc19
          Error: open /var/folders/j5/238lrgj54dz032np_n85c8nh0000gp/T/helmfile3497684751/foo/raw/0.1.0/raw-0.1.0.tgz1281403981: no such file or directory]
    snapshot_test.go:175:
        	Error Trace:	/Users/MB-873/Documents/helmfile/test/e2e/template/helmfile/snapshot_test.go:175
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestHelmfileTemplateWithBuildCommand/oci_chart_pull
        	Messages:   	Unable to run helmfile with args [-f testdata/snapshot/oci_chart_pull/input.yaml template]
```

This PR contains two commits:

1. Add failing E2E test to generate manifests from OCI based chart
2. Add missing `--untar` flag to `helm pull` command

### CODE REVIEW

File: test/e2e/template/helmfile/snapshot_test.go [hunk 1]
```diff
@@ -191,6 +243,22 @@ func execHelmPackage(t *testing.T, localChart string) string {
 	return strings.TrimSpace(tgzAbsPath)
 }
 
+// execHelmPush pushes helm package to oci based helm repository,
+// then returns its digest.
+func execHelmPush(t *testing.T, tgzPath, remoteUrl string) string {
```
Comment (yxxhero) [inline test/e2e/template/helmfile/snapshot_test.go:248]: return error will be better?

Comment (yxxhero) [reply test/e2e/template/helmfile/snapshot_test.go:248]: @toVersus

**Review by yxxhero — APPROVED** [review]
LGTM

File: test/e2e/template/helmfile/snapshot_test.go [hunk 2]
```diff
@@ -155,6 +177,37 @@ func TestHelmfileTemplateWithBuildCommand(t *testing.T) {
 
 			gotStr := string(got)
 			gotStr = strings.ReplaceAll(gotStr, fmt.Sprintf("chart=%s", wd), "chart=$WD")
+			// OCI based helm charts are pulled and exported under temporary directory.
+			// We are not sure the exact name of the temporary directory generated by helmfile,
+			// so redact its base directory name with $TMP.
+			if config.LocalDockerRegistry.Enabled {
```
Comment (mumoshu) [inline test/e2e/template/helmfile/snapshot_test.go:183]: Perfect! Awesome job @toVersus!

**Pull request merged**